adserver/staff/views.py

```diff
@@ -91,28 +92,39 @@ class PublisherPayoutView(StaffUserMixin, TemplateView):
     """
 
     template_name = "adserver/staff/publisher-payout-list.html"
+    # Two hours should be enough to run most payouts
+    CACHE_SECONDS = 7200
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        limit = int(self.request.GET.get("limit", 50))
+        limit = int(self.request.GET.get("limit", 0))
         all_publishers = self.request.GET.get("all")
         publisher_slug = self.request.GET.get("publisher")
 
         queryset = Publisher.objects.all()
         if publisher_slug:
             queryset = queryset.filter(slug__startswith=publisher_slug)
 
+        if limit:
+            queryset = queryset[:limit]
+
         payouts = {}
 
-        for publisher in queryset[:limit]:
-            data = generate_publisher_payout_data(publisher)
+        for publisher in queryset:
+            data = generate_publisher_payout_data(
+                publisher, include_current_report=False
+            )
+
+            # Cache payout data to make running payouts faster
+            data = cache.get(f"payout-{publisher.pk}")
```

Comment:
If we're just going to cache all the data, couldn't we just cache the whole view?

Reply:
We need to have the publisher payout state change over time as we update it. We could in theory do that in the template, but that seems pretty hacky.